### PR TITLE
fix: crash when restoring the app after a long period of inactivity

### PR DESCRIPTION
**Pull request by dixidroid**
Changed the way course data is loaded.

Previously there was the preloadCourseStructure function that was called when we opened a course. Then all data was cached and used in other places such as open content and etc. When the app was in the background for a long time all the data was cleared by the system and if the user tries to open the app - it crashes. 

I changed the way the Course Structure is retrieved and added a check for empty data.

https://github.com/openedx/openedx-app-android/issues/139

### CODE REVIEW

**Comment by omerhabib26** [comment]
Reviewing

**Comment by HamzaIsrar12** [comment]
Hey @dixidroid, just a heads-up, this PR will remain in a stale position until we merge #278, as it will update the underlying code fixed in this PR. 🕐

**Comment by HamzaIsrar12** [comment]
@dixidroid  The base PR has been merged. Could you please reassess the scope of this ticket? Thank You

File: course/src/main/java/org/openedx/course/data/repository/CourseRepository.kt [hunk 1]
```diff
@@ -26,35 +29,29 @@ class CourseRepository(
...
-    }
+        if (networkConnection.isOnline()) {
+            val response = api.getCourseStructure(
+                "stale-if-error=0",
+                "v3",
+                preferencesManager.user?.username,
+                courseId
+            )
+            courseDao.insertCourseStructureEntity(response.mapToRoomEntity())
+            courseStructure[courseId] = response.mapToDomain()
```
Comment (volodymyr-chekyrta) [inline course/src/main/java/org/openedx/course/data/repository/CourseRepository.kt:44]: Please remove the extra line

File: course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt [hunk 2]
```diff
@@ -248,7 +243,7 @@ class CourseContainerViewModel(
     fun updateData() {
         viewModelScope.launch {
             try {
-                interactor.preloadCourseStructure(courseId)
+                interactor.getCourseStructure(courseId)
```
Comment (volodymyr-chekyrta) [inline course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt:246]: Will it actually update the data, or will it just return the last cached course?

**Comment by volodymyr-chekyrta** [comment]
The overall approach looks good.
Concern that the `updateData` function just returns cached data instead of reloading.

**Comment by omerhabib26** [comment]
Resuming the review

**Review by omerhabib26 — DISMISSED** [review]
LGTM

File: course/src/main/java/org/openedx/course/data/repository/CourseRepository.kt (hunk 1 above)
Comment (dixidroid) [reply course/src/main/java/org/openedx/course/data/repository/CourseRepository.kt:48]: done

File: course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt (hunk 2 above)
Comment (dixidroid) [reply course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt]: no, but I changed how it works and right now it should update the data )

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**